### PR TITLE
Bump go to 1.17

### PR DESCRIPTION
**Pull request by ryantking**
**What this PR does / why we need it**:

Bumps go 1.16 -> 1.17 and k8s.io/... deps 1.22.2 -> 1.22.3

**Which issue(s) this PR fixes**:

Part of https://github.com/kubernetes-sigs/kubebuilder/issues/2421

**Special notes for your reviewer**:

**Additional documentation**:

### CODE REVIEW

File: examples/guestbook-operator/go.mod [hunk 1]
```diff
@@ -10,4 +10,110 @@ require (
 	sigs.k8s.io/kubebuilder-declarative-pattern v0.0.0-20210922163802-cac4a6cf1977
 )
 
+require (
```
Comment (camilamacedo86) [inline examples/guestbook-operator/go.mod:13]: /hold 

we ought to not have twice the required
we need to remove the indirect ones. 
to upgrade the go version we ought to not change the deps.

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Thank you for your contribution 🥇 
Please, see; https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/189/files#r751483417

**Review by camilamacedo86 — APPROVED** [review]

**Comment by atoato88** [comment]
It's LGTM also.
Sorry for late response.

**Pull request merged**